torch/distributed/_tensor/api.py

```diff
@@ -105,16 +105,19 @@ def forward(  # type: ignore[override]
         device_mesh: DeviceMesh,
         placements: Tuple[Placement, ...],
         run_check: bool,
+        shape: Optional[torch.Size] = None,
+        stride: Optional[Tuple[int, ...]] = None,
     ) -> "DTensor":
         ctx.previous_placement = placements
         ctx.previous_device_mesh = device_mesh
 
-        # if it's not by default run_check, we assume user is certain that each
-        # rank has the same tensor shape, and we just use that to calculate the
-        # global shape
-        tensor_shape, tensor_stride = compute_global_tensor_info(
-            input, device_mesh, placements
-        )
+        if not shape or not stride:
+            # if it's not by default run_check, we assume user is certain that each
+            # rank has the same tensor shape, and we just use that to calculate the
+            # global shape
+            tensor_shape, tensor_stride = compute_global_tensor_info(
+                input, device_mesh, placements
+            )
```

Comment:
can we just do:
```
else:
    tensor_shape=shape
    tensor_stride=stride
```
so that we don't need to check them again in the later callsites